gnosis/safe-cmm-app-react
# Feature/#7/funding comp load token from web3provider

# Description

- Loading token from Web3Provider context.
- Validating input onChange.

![screenshot_2020-08-18_13-38-38](https://user-images.githubusercontent.com/43217/90563085-20177000-e158-11ea-8648-1986e600fa79.png)


# To Test
1. Run storybook locally: `npm run storybook`
2. Visit `http://localhost:6006/?path=/story/fundinginput--default`

- [ ] Check mock token `TKN` is displayed on all `FundingInput` stories
- [ ] Verify input accepts ONLY numbers
- [ ] Input turns red when trying to insert invalid numbers
- [ ] Input is back to normal once valid number is inserted

# Background

Loading token info async from the Web3Provider context. It causes a little flicker I'm not a big fan of.
I'm not happy with this solution and would prefer to instead load the token from parent component and pass it down the `tokenDetails` object instead.

Thoughts?
